tests/test_protocol.py

```diff
@@ -1700,3 +1702,309 @@ async def merge_dicts_impl(self, one: Dict[str, Any], other: Dict[str, int], any
     response: Result = await client.merge_dicts(one, other, None)
     assert response.code == 200
     assert response.result == {"data": {**one, **other}}
+
+
+@pytest.mark.parametrize(
+    "param_type,param_value,expected_url,expected_status_code",
+    [
+        (
+            Dict[str, str],
+            {"a": "b", "c": "d", ",&?=%": ",&?=%"},
+            "/api/v1/test/1/monty?filter.a=b&filter.c=d&filter.,&?=%=,&?=%",
```

Comment:
How do you determine whether the last filter evaluates to:
* key="`,&?=%=,&?`"; value="`%`"
OR
* key="`,&?=%`"; value="`,&?=%`"
OR 
* key="`,&?`"; value="`%=,&?=%`"

Reply:
You are correct, that by looking at this, you can't.
In the test case however, the url is decoded (or unquoted), before making this comparison, because I thought that it would be easier to read this way, but now I'm not sure that's the case. 
Do you think I should change the `expected_url` to the url encoded version?

Reply:
I think that makes sense. By doing that we can visually see what the test case is verifying.